### PR TITLE
[KBFS-2747] Fix some things that were broken on mobile

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

File: shared/fs/index.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ class Files extends React.PureComponent<FolderProps> {
     const {path, items, progress} = this.props
     const list =
       progress === 'pending' ? (
-        <List items={[null, null, null]} renderItem={this._renderRowPlaceholder} />
+        <List items={['1', '2', '3']} renderItem={this._renderRowPlaceholder} />
```
Comment (songgao) [inline shared/fs/index.js]: Not that I'm against it, but I'm curious why this was necessary?

Comment (jzila) [inline shared/fs/index.js]: I think if you only pass one parameter to the iterator, it takes the item, not the index. It errored until I fixed it.

Comment (songgao) [reply shared/fs/index.js]: Hmm https://github.com/coderiety/react-list#itemrendererindex-key doesn't mention so. Also how did it work on desktop?

Comment (jzila) [reply shared/fs/index.js]: ¯\_(ツ)_/¯

Comment (songgao) [reply shared/fs/index.js]: Ugh it's bothering me that I can't understand this. I just tried some `console.log` and (on desktop) the `index` was indeed the index even as the only parameter. I'll try mobile sometime I guess.

Comment (jzila) [reply shared/fs/index.js]: This is probably why:
https://reactjs.org/docs/lists-and-keys.html
```
    <li key={number.toString()}>
      {number}
    </li>
```

Comment (songgao) [reply shared/fs/index.js]: Hmm I don't understand. This is not the list we are using right?

Comment (jzila) [reply shared/fs/index.js]: Strings vs ints. You had ints before. Now they are strings.

Comment (songgao) [reply shared/fs/index.js]: Here's what I found:

1. We are using [`FlatList`](https://facebook.github.io/react-native/docs/flatlist.html) on mobile, and [`react-list`](https://github.com/coderiety/react-list) on desktop.

2. `FlatList` uses `renderItem({ item: Object, index: number, ...}), while `react-list` expects `itemRenderer(index, key)`.

3. We have an adapter for each: [desktop](https://github.com/keybase/client/blob/a93667c51dc7fee080b042a5ba172f945913dd83/shared/common-adapters/list.desktop.js) and [mobile](https://github.com/keybase/client/blob/a93667c51dc7fee080b042a5ba172f945913dd83/shared/common-adapters/list.native.js), and they both accept the same form which is `renderItem(index, item)`.

I think what makes the difference might be `FlatList` is treating `null` items differently. I'll try to figure it out tomorrow!

Comment (jzila) [reply shared/fs/index.js]: Ah it's still an int because it's passing the index.

**Pull request merged**